numpy/core/fromnumeric.py

```diff
@@ -52,17 +52,19 @@ def _wrapit(obj, method, *args, **kwds):
 
 
 def _wrapfunc(obj, method, *args, **kwds):
-    try:
-        return getattr(obj, method)(*args, **kwds)
-
-    # An AttributeError occurs if the object does not have
-    # such a method in its class.
+    bound = getattr(obj, method, None)
+    if bound is None:
+        return _wrapit(obj, method, *args, **kwds)
 
-    # A TypeError occurs if the object does have such a method
-    # in its class, but its signature is not identical to that
-    # of NumPy's. This situation has occurred in the case of
-    # a downstream library like 'pandas'.
-    except (AttributeError, TypeError):
+    try:
+        return bound(*args, **kwds)
+    except TypeError:
+        # A TypeError occurs if the object does have such a method in
+        # its class, but its signature is not identical to that of
+        # NumPy's. This situation has occurred in the case of a downstream
+        # library like 'pandas'.
+        #
+        # In this case, we want to traceback-chain a potential exception.
```

Comment:
I noticed that the commit message doesn't quite fit policy: It should have a `MAINT: ` in front.  Could you change it? (Not sure how to do it as a maintainer.)

And that is perhaps just enough reason to also rewrite the comment here: "In this case," now reads oddly. Maybe something like: "Call _wrapit from within the except clause to ensure a potential exception has a traceback chain".

With that done, I'll merge...

Reply:
There is an Edit button next to the PR title. Note you can also change the merge message when you press the "Merge Pull Request" button

Reply:
both points handled

Reply:
@mattip - that changes the PR title or the merge commit title, but not the commit with the actual work (which I think is the one we want...).

@anntzer - thanks for the update, will now merge.